edge-util/src/Microsoft.Azure.Devices.Edge.Util/Retry.cs

```diff
@@ -3,44 +3,56 @@
 namespace Microsoft.Azure.Devices.Edge.Util
 {
     using System;
+    using System.Threading;
     using System.Threading.Tasks;
 
     public static class Retry
     {
+        static async Task<Option<T>> DoOnce<T>(
```

Comment:
Should we call Do<T> with zero retry interval and none cancellation token in DoOnce<T> method?